src/main/java/sg/ncl/MainController.java

```diff
@@ -5116,4 +5130,199 @@ private TeamQuota extractTeamQuotaInfo(String responseBody) {
     private String getExperimentMessage(String expName, String teamName) {
         return "Experiment " + expName + " in team " + teamName;
     }
+
+    @RequestMapping(value="/add_member/{teamId}", method= RequestMethod.GET)
+    public String addMember(@PathVariable String teamId, Model model) {
+        model.addAttribute("addMemberForm", new AddMemberForm());
+        return "add_member";
+    }
+
+    @RequestMapping(value="/add_member/{teamId}", method= RequestMethod.POST)
+    public String addMember(@PathVariable String teamId, @Valid AddMemberForm addMemberForm,
+                            final RedirectAttributes redirectAttributes)  throws WebServiceRuntimeException {
+        log.info("Starting to adding members by email for team {}", teamId);
+
+        String emails[] = addMemberForm.getEmails().split("\\r?\\n");
+
+        JSONArray jsonArray = new JSONArray();
+        for (int i = 0; i< emails.length; i++){
+            if (!VALID_EMAIL_ADDRESS_REGEX.matcher(emails[i]).matches()) {
+                addMemberForm.setErrMsg(EMAIL_ADDRESS_IS_NOT_VALID);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the array designator from the variable to the type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1197)